src/python/CRABClient/JobType/Analysis.py

```diff
@@ -92,14 +92,11 @@ def run(self, filecacheurl = None):
         isbchecksum = uploadResults[1]
 
         # Upload list of user-defined input files to process as the primary input
-        userFileName = getattr(self.config.Data, 'userInputFile', None)
-        if userFileName:
-            self.logger.debug("Attaching a list of user-specified primary input files from %s." % userFileName)
-            fnames = []
-            for fname in open(userFileName).readlines():
-                fnames.append(fname.strip())
-            configArguments['userfiles'] = filter(lambda x: x, fnames) #removing whitelines and empty objects
-
+        userFilesList = getattr(self.config.Data, 'userInputFiles', None)
```

Comment:
It makes sense to me. Unless there are use cases where people want to run two jobs on the same input file... In that case we should just print a warning. but I can't see a use case for this, so let's make it a set and print a warning.